src/courtroom.cpp

```diff
@@ -3994,38 +4117,62 @@ void Courtroom::set_hp_bar(int p_bar, int p_state)
   if (p_state < 0 || p_state > 10)
     return;
 
+  int prev_state = p_state;
   if (p_bar == 1) {
     ui_defense_bar->set_image("defensebar" + QString::number(p_state));
+    prev_state = defense_bar_state;
     defense_bar_state = p_state;
   }
   else if (p_bar == 2) {
     ui_prosecution_bar->set_image("prosecutionbar" + QString::number(p_state));
+    prev_state = prosecution_bar_state;
     prosecution_bar_state = p_state;
   }
+
+  QString sfx_name;
+  QString effect_name;
+  if (p_state > prev_state) {
+    sfx_name = ao_app->get_penalty_value("recovery");
+    effect_name = ao_app->get_penalty_value("recovery_effect").toLower();
+  }
+  else if (p_state < prev_state) {
+    sfx_name = ao_app->get_penalty_value("deterioration");
+    effect_name = ao_app->get_penalty_value("deterioration_effect").toLower();
```

Comment:
Let's stick to simple names, e.g. `hp_increased_{sfx|effect}`, `hp_decreased_{sfx|effect}`. It keeps the terms consistent since we don't use "recovery" or "deterioration" anywhere else.